snakefile.py

```diff
@@ -218,20 +240,23 @@ def tool(name):
 
 
 rule fastqc_raw:
-    input: os.path.join(READS_DIR, "{sample}_R{read_num}.fastq")
+    input: trim_reads_input
     output:
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.zip'),
-        os.path.join(FASTQC_DIR, '{sample}', '{sample}_R{read_num}_fastqc.html')
-    log: os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R{read_num}.log')
+        r1_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R1_fastqc.html'),
+        r2_rep = os.path.join(FASTQC_DIR, '{sample}', '{sample}_R2_fastqc.html')
+    log: [os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R1.log'), os.path.join(LOG_DIR, 'fastqc_{sample}_raw_R2.log')]
```

Comment:
will there always be two files?  In this case: why use trim_reads_input?

Reply:
In my understanding - trim_reads_input first and foremost matches the read-files to the corresponding sample name as given by the sample sheet. 
And yes this is either one or two files. 


Yes - later when I started to think about how to make the pipeline usable with single-end input this is one of the points I stumbled over.[ My first approach](https://github.com/BIMSBbioinfo/pigx_sarscov2_ww/commit/cc811702b9b914e1f947076f21af2cc534c976c3) was just to write one rule for paired-end and one rule for single-end. I think using map_input that could be handled more elegantly - but then I wasn't sure about how to make the outputs dynamic too. Because I think with single-end the output would be just `{sample}_fastqc.zip`